watershed-android/watershed/app/src/main/java/com/blueprint/watershed/Views/CircleImageView.java

```diff
@@ -0,0 +1,373 @@
+        if (this.isSuperFaded()) {
+            this.setAlpha(UNSELECTABLE_ALPHA / 255f);
+        } else if ((this.isSelectEnabled() && !this.isSelected()) || this.isFaded()) {
+            this.setAlpha(UNFOCUSED_ALPHA / 255f);
+        } else {
+            this.setAlpha(1f);
+        }
+
+        mBitmap = bitmap;
+        //mBitmap = bitmap.copy(Bitmap.Config.ARGB_8888, true);
+    }
+
+    /**
+     * Setter and getter for isDraggable.
+     */
+    public static void setDraggable(boolean bool) { isDraggable = bool; }
+
+    public static boolean isDraggable() { return isDraggable; }
+
+    /**
+     * Setter and getter for isSelectEnabled.
+     */
+    public void setSelectEnabled(boolean bool) {
+        isSelectEnabled = bool;
+        setSoundEffectsEnabled(bool);
+    }
+
+    public boolean isSelectEnabled() { return isSelectEnabled; }
+
+    /**
+     * Setter and getter for isSelected.
+     */
+    public void setSelected(boolean bool) {
+        isSelected = bool;
+        isStateChanged = true;
+    }
+
+    public boolean isSelected() { return isSelected; }
+
+    /**
+     * Setter and getter for isFaded.
+     */
+    public void setFaded(boolean bool) {
+        this.isFaded = bool;
+        this.isStateChanged = true;
+    }
+
+    public boolean isFaded() { return this.isFaded; }
+
+    /**
+     * Setter and getter for isSuperFaded.
+     */
+    public void setSuperFaded(boolean bool) {
+        this.isSuperFaded = bool;
+        this.isStateChanged = true;
+    }
+
+    public boolean isSuperFaded() { return this.isSuperFaded; }
+
+    /**
+     * Setter and getter for mText.
+     */
+    public void setText(String text) {
+        mText = text;
+        isStateChanged = true;
+    }
+
+    public String getText() { return mText; }
+
+    /**
+     * Getter for isStateChanged.
+     */
+    public boolean isStateChanged() { return isStateChanged; }
+
+    /**
+     * Override setImageDrawable to toggle isStateChanged.
+     * @param drawable
+     */
+    @Override
+    public void setImageDrawable(Drawable drawable) {
+        super.setImageDrawable(drawable);
+        isStateChanged = true;
+    }
+
+    public Drawable getImageDrawable() {
+        return getDrawable();
+    }
+
+    /**
+     * Override setImageResource to toggle isStateChanged.
+     *
+     * DO NOT USE THIS METHOD!
+     *
+     * Deprecated because:
+     * http://developer.android.com/reference/android/widget/ImageView.html#setImageResource%28int%29
+     * "This does Bitmap reading and decoding on the UI thread, which can cause a latency hiccup.
+     * If that's a concern, consider using setImageDrawable(android.graphics.drawable.Drawable) or
+     * setImageBitmap(android.graphics.Bitmap) and BitmapFactory instead."
+     * @param resId
+     */
+    @Override
+    public void setImageResource(int resId) {
+        super.setImageResource(resId);
+        isStateChanged = true;
+    }
+
+}
```

Comment:
hehe reduce, reuse recycle. Can you go through this and clean up the commented code?